apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -84,22 +87,49 @@ export default class LibraryPublisher extends React.Component {
     this.setState({libraryName: sanitizedName});
   };
 
-  publish = () => {
-    const {libraryDescription, libraryName, selectedFunctions} = this.state;
-    const {librarySource, sourceFunctionList} = this.props.libraryDetails;
-    const {libraryClientApi, onPublishSuccess} = this.props;
-    const functionsToPublish = sourceFunctionList.filter(sourceFunction => {
+  getFunctionsToPublish = () => {
+    const {selectedFunctions} = this.state;
+    const {sourceFunctionList} = this.props.libraryDetails;
+    return (sourceFunctionList || []).filter(sourceFunction => {
       return selectedFunctions[sourceFunction.functionName];
     });
+  };
 
-    if (!(libraryDescription && functionsToPublish.length > 0)) {
+  validateAndPublish = async () => {
+    const {libraryDescription, libraryName} = this.state;
+
+    if (!(libraryDescription && this.getFunctionsToPublish().length > 0)) {
       this.setState({publishState: PublishState.INVALID_INPUT});
       return;
     }
 
+    // Validate library name/description input for profanity before publishing.
+    try {
+      const profaneWords = await this.props.findProfanity(
+        `${libraryName} ${libraryDescription}`
+      );
+      if (profaneWords) {
```

Comment:
Is profaneWords `[]` if there are none? I think `!![]` is true, which would make you always hit this case

Reply:
nope, the endpoint returns `null` if there are no profane words